src/ds/concept.h

```diff
@@ -38,5 +38,40 @@ namespace snmalloc
   template<typename T, typename U>
   concept ConceptSame = std::is_same<T, U>::value;
 #  endif
+
+  /**
+   * Equivalence mod std::remove_reference
+   */
+  template<typename T, typename U>
+  concept ConceptSameModRef = ConceptSame<std::remove_reference_t<T>, U>;
```

Comment:
Why is this not symmetric?

```suggestion
  concept ConceptSameModRef = ConceptSame<std::remove_reference_t<T>, std::remove_reference_t<U>>;
```

Reply:
Because `U` is provided by the `concept` `requires` expression, as in `{ E } -> ConceptSameModRef<U>`, and it seemed silly to specify `U &` only to remove it.  But, I suppose, `U` might be derived from some other thing rather than named explicitly, so sure.